### ISSUE TITLE
Create a hook for working with siblings elements

### ISSUE THREAD

**Issue by dgdavid** [openSUSE/agama#565]
In #563 we introduced an internal function to work with siblings in the core/Sidebar component. It works with the DOM API directly and it works. However, would be nice to extract it to a hook, which is a bit more _React alike_ and will allow (probably among other benefits):

* Better reusability
* Better testing
* A central place for implementing a logic to be polite and restore the previous attributes values if they were present instead of simply removing them.

**Comment by balsa-asanovic** [comment]
Hey there, can I try to work on this?

**Comment by imobachgs** [comment]
Hi, @balsa-asanovic. Sure, you can work on this one if you wish. Thanks in advance!

**Comment by balsa-asanovic** [comment]
I am trying to run the project using ./setup.sh script, but I am running into problems.

Could you help me? @imobachgs 

Here's the screenshot:

![image](https://github.com/openSUSE/agama/assets/112288843/f298abfe-e9df-44e0-8617-9e307aa27f68)

**Comment by imobachgs** [comment]
Hi @balsa-asanovic,

Let's try. It looks like the D-Bus service is not running. Did you get any errors when running the `setup.sh` script? You could try to run the service from sources by typing:

```sh
cd service
sudo bundle exec bin/agama
```

It will take a few seconds to boot.

**Comment by balsa-asanovic** [comment]
Hey @imobachgs,

Thanks for the help.

I don't actually get any errors when running `setup.sh` script.

However, when running `sudo bundle exec bin/agamactl`, some issues do pop-up:

![image](https://github.com/openSUSE/agama/assets/112288843/c311c1b2-23df-42d6-b189-76c6b3a0200e)
![image](https://github.com/openSUSE/agama/assets/112288843/8f675181-ee60-4c65-ae45-a8da4bbf8e77)

**Comment by balsa-asanovic** [comment]
Okay, I've just did that, so now when I run the service it seems to be issues free, but the initial problem remains:

![image](https://github.com/openSUSE/agama/assets/112288843/16086a63-620d-4f12-adc2-690a5a6c7615)

**Comment by balsa-asanovic** [comment]
Yeah, that's right. After I run cockpit.logout() I get logged out. When I log back in again, I get the same message

![image](https://github.com/openSUSE/agama/assets/112288843/06586285-2f11-48eb-809d-e5e91b5d834c)
![image](https://github.com/openSUSE/agama/assets/112288843/5629d31e-f575-4c0a-bb2a-4bfed1e4ad5b)

**Comment by balsa-asanovic** [comment]
Okay guys, I feel dumb as usually when the problem gets resolved :sweat_smile: 

Now, I logged in as root and it seems the everything works fine, previously I used my regular user account.

![image](https://github.com/openSUSE/agama/assets/112288843/41beb1a1-a0a1-4f34-9d67-22dd2ba95002)

**Comment by imobachgs** [comment]
Oh, good catch. I think it is something we should make clear in the documentation.

**Comment by dgdavid** [comment]
> Okay guys, I feel dumb as usually when the problem gets resolved sweat_smile
> 

No worries, as @imobachgs said we should clarify it in the documentation


> Thank a lot for the help both to @imobachgs and @dgdavid

You're more than welcome.

:warning: **PLEASE** remind that Agama will delete all the content on the disk selected for the installation. In the screenshot above I see Agama telling you _"Install using device /dev/sda, 111.79 GiB, **Windows** and deleting all its content"_ We highly recommend you using a virtual machine for testing and develop instead of a real one.

**Comment by balsa-asanovic** [comment]
> :warning:  PLEASE remind that Agama will delete all the content on the disk selected for the installation. In the screenshot above I see Agama telling you "Install using device /dev/sda, 111.79 GiB, Windows and deleting all its content". We highly recommend you using a virtual machine for testing and develop instead of a real one.

Of course, thanks for the reminder, I'll set up virtual environment for further tinkering.

**Comment by balsa-asanovic** [comment]
Hey @dgdavid, I've submitted some code changes through #611 

Check it out when you find the time, I am sure there will be a few issue for me to unpack. :smile: 

Thanks.
